### PR TITLE
test(consumer): handle transaction events

### PR DESCRIPTION
**Pull request by dvdblk**

### CODE REVIEW

File: src/data_collection/tests/unit/test_consumer.py [hunk 1]
```diff
@@ -282,84 +285,839 @@ async def test_transfer_fungible_event_inserted(
...
+            tx_receipt_data=transaction_receipt_data,
+            w3_block_hash=Mock(),
+        )
+
+        # Assert
+        consumer.db_manager.insert_transaction_logs.assert_awaited_once_with(
+            **transaction_logs_data.dict()
+        )
+        consumer.db_manager.insert_contract_supply_change.assert_not_awaited()
+        consumer.db_manager.insert_pair_liquidity_change.assert_awaited_once()
```
Comment (dvdblk) [inline src/data_collection/tests/unit/test_consumer.py:838]: needs `assert_awaited_once_with` for `amount0` and `amount1`

File: src/data_collection/tests/unit/test_consumer.py [hunk 2]
```diff
@@ -282,84 +285,839 @@ async def test_transfer_fungible_event_inserted(
...
+            tx_data=transaction_data,
+            tx_receipt=Mock(),
+            tx_receipt_data=transaction_receipt_data,
+            w3_block_hash=Mock(),
+        )
+
+        # Assert
+        consumer.db_manager.insert_transaction_logs.assert_not_awaited()
+        consumer.db_manager.insert_contract_supply_change.assert_not_awaited()
+        consumer.db_manager.insert_pair_liquidity_change.assert_not_awaited()
```
Comment (dvdblk) [inline src/data_collection/tests/unit/test_consumer.py:891]: ~~needs `assert_awaited_once_with` for `amount0` and `amount1`~~

Comment (dvdblk) [reply src/data_collection/tests/unit/test_consumer.py:891]: nevermind, it doesn't

File: src/data_collection/tests/unit/test_consumer.py [hunk 3]
```diff
@@ -282,84 +285,839 @@ async def test_transfer_fungible_event_inserted(
...
+            tx_receipt_data=transaction_receipt_data,
+            w3_block_hash=Mock(),
+        )
+
+        # Assert
+        consumer.db_manager.insert_transaction_logs.assert_awaited_once_with(
+            **transaction_logs_data.dict()
+        )
+        consumer.db_manager.insert_contract_supply_change.assert_not_awaited()
+        consumer.db_manager.insert_pair_liquidity_change.assert_awaited_once()
```
Comment (dvdblk) [inline src/data_collection/tests/unit/test_consumer.py:948]: needs `assert_awaited_once_with` for `amount0` and `amount1`

File: src/data_collection/tests/unit/test_consumer.py [hunk 4]
```diff
@@ -282,84 +285,839 @@ async def test_transfer_fungible_event_inserted(
...
```
Comment (dvdblk) [inline src/data_collection/tests/unit/test_consumer.py:1063]: `assert_awaited_once_with` correct values

**Review by dvdblk — COMMENTED** [review]
Pls add a test where `TransferFungibleEvent` to zero address is returned from `get_tx_events`, but only `BurnFungibleEvent` is present in config.
